### PR TITLE
PERF: Speed-up taxonomic string preprocessing

### PR DESCRIPTION
**Pull request by ElDeveloper**
The string operations are vectorized which makes processing the EMP
taxonomy strings go from 45 seconds to ~1.5 seconds.

### CODE REVIEW

**Comment by emperor-helper** [comment]
The following artifacts were built for this PR: [empire-biplot.qzv](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fmchelper.ucsd.edu%2Fdownloads%2Fempress%2F326%2Fempire-biplot.qzv), [empire.qzv](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fmchelper.ucsd.edu%2Fdownloads%2Fempress%2F326%2Fempire.qzv), [empress-tree.qzv](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fmchelper.ucsd.edu%2Fdownloads%2Fempress%2F326%2Fempress-tree.qzv)

File: empress/taxonomy_utils.py [hunk 1]
```diff
@@ -100,11 +100,19 @@ def split_taxonomy(feature_metadata):
...
+        # Split the single column of taxonomy strings into n columns, where n
+        # is the highest number of taxonomic levels in any string. We have to
+        # account for leading and trailing whitespace (.str.strip()) as well as
+        # for whitespace between delimiters (.str.split()). The regular
+        # expression splits tokens separated by semicolons and up to one
+        # leading or trailing space. Importantly, using the expand keyword
+        # ensures that all rows are padded to have n tokens. The final step is
+        # to replace those empty values with "Unspecified".
+        tax_levels = feature_metadata[tax_col_name]\
+            .str.strip().str.split(r'\s{0,1};\s{0,1}', expand=True)
```
Comment (fedarko) [inline empress/taxonomy_utils.py:112]: If I understand correctly, the regex `\s{0,1}` means 0 or 1 instances of a whitespace character. Could we replace these with `\s*`? (basically, would a silly dataset with like 20 tabs on one side of a semicolon cause problems here?)

I tried this out in python and it seems to work ok on such a dataset, but ... I'm not sure why this is the case. Less of a suggestion, more of a question :)

**Comment by fedarko** [comment]
Looks good to me! Thanks so much @ElDeveloper :)

**Comment by ElDeveloper** [comment]
Thanks @fedarko!

**Pull request merged**